### PR TITLE
 Adapt things to new blob metadata API 

### PR DESCRIPTION
**Pull request by millerdev**
Review commits individually 🐠

@snopoke @sravfeyn

### CODE REVIEW

**Comment by millerdev** [comment]
Hold off on review, going to rebase on master and fix the merge conflicts.

**Comment by millerdev** [comment]
Fixed.

File: corehq/blobs/tests/test_metadata.py
Comment (snopoke) [inline corehq/blobs/tests/test_metadata.py]: does this happen somewhere else now?

File: corehq/ex-submodules/casexml/apps/phone/restore.py
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: this will hash all these to the same SQL DB which could create a hotspot for ICDS. What about using the restore user ID?

File: corehq/apps/ota/migrations/0006_update_blob_paths.py
Comment (snopoke) [inline corehq/apps/ota/migrations/0006_update_blob_paths.py]: took me a while to figure out that `_default` was the bucket name under the old API. Adding that to the comment could help future readers.

**Review by snopoke — COMMENTED** [review]
Generally fine but would want @proteusvacuum to give a go-head on the `DataFile` changes before merging. Also the question about sharding restore cache on domain.

File: corehq/blobs/metadata.py
Comment (millerdev) [reply corehq/blobs/metadata.py]: I want arguments to this method to always be passed as keyword arguments so it's clear what's being passed in. Do you think that's too strict/unnecessary/something else?

File: corehq/blobs/tests/test_metadata.py
Comment (millerdev) [reply corehq/blobs/tests/test_metadata.py]: Yes, it's done by `TemporaryFilesystemBlobDB.clean_db()` which is called by `close()`

https://github.com/dimagi/commcare-hq/blob/0901a7841bdbe5ba4a5e08b5bf15009e38bda8d2/corehq/blobs/tests/util.py#L95-L98

File: corehq/apps/ota/migrations/0006_update_blob_paths.py
Comment (millerdev) [reply corehq/apps/ota/migrations/0006_update_blob_paths.py]: 0b0d406f3f96f9667f75d227d8df70f1be1d0f2e

File: corehq/apps/export/models/new.py
Comment (millerdev) [reply corehq/apps/export/models/new.py]: 468e594ea0c43c9d113634503500fc9a4d41a141

File: corehq/ex-submodules/casexml/apps/phone/restore.py
Comment (millerdev) [reply corehq/ex-submodules/casexml/apps/phone/restore.py]: Good idea. 7ed7942c01c8cc4ee7a62630d5d7afc5a601c394

File: corehq/apps/export/migrations/0005_datafile_blobmeta.py
Comment (snopoke) [reply corehq/apps/export/migrations/0005_datafile_blobmeta.py]: Offlined with @millerdev

File: corehq/blobs/metadata.py
Comment (snopoke) [reply corehq/blobs/metadata.py]: I suppose that's OK

**Review by snopoke — APPROVED** [review]

**Comment by proteusvacuum** [comment]
:+1: looks like the recent functionality I added to `DataFile` was retained.

**Pull request merged**
